docarray/utils/create_dynamic_doc_class.py

```diff
@@ -51,6 +51,8 @@ class MyDoc(BaseDoc):
     """
     fields: Dict[str, Any] = {}
     for field_name, field in model.__annotations__.items():
+        if field_name not in model.__fields__:
+            continue
         field_info = model.__fields__[field_name].field_info
```

Comment:
FWIW: It might be worth using something like `pydash.get` here in case traversal breaks at a different location.